c7n/resources/sagemaker.py

```diff
@@ -107,6 +107,53 @@ def _augment(j):
             return list(filter(None, w.map(_augment, jobs)))
 
 
+@resources.register('sagemaker-transform-job')
+class SagemakerTransformJob(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_transform_jobs', 'TransformJobSummaries', None)
+        detail_spec = (
+            'describe_transform_job', 'TransformJobName', 'TransformJobName', None)
+        id = 'TransformJobArn'
+        name = 'TransformJobName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = None
+
+    permissions = (
+        'sagemaker:ListTransformJobs', 'sagemaker:DescribeTransformJobs',
+        'sagemaker:ListTags')
```

Comment:
rather than use sagemaker list tags use the universal augment in tags.py as the resource group tagging can address batches of 20 vs retrieving tags per individual resource.